File: app/lib/models/actor.dart
import 'package:freezed_annotation/freezed_annotation.dart';
import 'package:wanderer/models/record.dart';

part 'actor.freezed.dart';
part 'actor.g.dart';

@Freezed()
abstract class Actor with _$Actor, RecordFunctions implements IRecord {
  const Actor._();

  const factory Actor({
    required String id,
    required String collectionId,
    required String collectionName,
    required DateTime created,
    required DateTime updated,
    required String username,
    @JsonKey(name: 'preferred_username') required String preferredUsername,
    String? domain,
    String? summary,
    String? published,
    int? followerCount,
    int? followingCount,
    required String iri,
    required String inbox,
    String? outbox,
    String? icon,
    String? followers,
    String? following,
    @Default(false) bool isLocal,
    @JsonKey(name: 'public_key') required String publicKey,
    @JsonKey(name: 'last_fetched') required String lastFetched,
    required String user,
  }) = _Actor;

  factory Actor.fromJson(Map<String, dynamic> json) => _$ActorFromJson(json);
}

File: web/src/lib/models/activitypub/actor.ts
import type { User } from "../user";

export interface Actor {
    id?: string;
    username: string;
    preferred_username: string;
    domain?: string;
    summary?: string;
    published?: string;
    follower_count?: number,
    following_count?: number,
    iri: string;
    inbox: string;
    outbox?: string;
    icon?: string;
    followers?: string;
    following?: string;
    is_local: boolean;
    public_key: string;
    last_fetched: string;
    user?: string
    expand?: {
        user?: User
    }
}

export interface ActorSearchResult {
    id: string;
    username: string;
    preferred_username: string;
    domain: string;
    is_local: boolean;
    iri: string;
    icon?: string;
}